### ISSUE TITLE
Cursor Color?

### ISSUE THREAD

**Issue by tgoodlet** [xonsh/xonsh#2088]
What's the recommended way to modify the cursor color?
I can change it so far with:
```python
>>> print('\033]12;red\007')
>>> print('\033]12;gray\007')
```

When I use this directly in my `.xonshrc` it works fine but as soon as I try to use it in a `$PROMPT` function (of course with `$UPDATE_PROMPT_ON_KEYPRESS = True`) I run into strange prompt clobbering issues. Using `printf` (subprocess mode) seems to work better but is very slow.

Any suggestions?

**Comment by melund** [comment]
You need to use the built-in system for customizing the prompt. You can
color it by adding a color keywords like this `{GREEN}`, it also supports
hex colors ( although you terminal may not,)

Den fre. 13. jan. 2017 09.24 skrev goodboy <notifications@github.com>:

**Comment by melund** [comment]
Hmm... Pressed send too early. You can find more on prompt formatting here:
http://xon.sh/tutorial.html#customizing-the-prompt

Den fre. 13. jan. 2017 09.36 skrev Morten Enemark Lund <melund@gmail.com>:

**Comment by tgoodlet** [comment]
@melund yes I've read through all of that and still haven't been able to use any of it to modify the cursor color. Maybe you can suggest how to do to that using `$PROMPT`?

**Comment by melund** [comment]
Sorry. I was a little to fast. I thought you were talking about the prompt.
I am not sure about customizing the cursor. Not even sure it possible now.
A PR for that would be great.

Den fre. 13. jan. 2017 09.45 skrev goodboy <notifications@github.com>:

**Comment by laerus** [comment]
this seems like a nice feature, i don't think prompt toolkit provides such functionality,
can you please report the errors you get when `$UPDATE_PROMPT_ON_KEYPRESS = True`,
a PR on this matter would be very appreciated!

**Comment by tgoodlet** [comment]
@laerus ok I've put it in my backlog and I'll do a little research hopefully this week.
I'll also get back to you with my `$PROMPT` function issues. I'm assuming a separate issue which links to this one is best?

**Comment by jnoortheen** [comment]
It is possible when the ansi codes are escaped (ptk shell)
<img width="631" alt="image" src="https://user-images.githubusercontent.com/6702219/153224730-3d45acfb-92e6-4e60-a150-b47c2658dadc.png">
